pkg/controller/test_helper.go

```diff
@@ -65,6 +67,38 @@ func readFile(filename string) (string, error) {
 	return str, nil
 }
 
+// WaitLoadtestFunc handles conditional watch events
+func WaitLoadtestFunc(event watch.Event) (bool, error) {
```

Comment:
Should this be private? since it's only used by `WaitLoadtest`
```suggestion
func waitLoadtestFunc(event watch.Event) (bool, error) {
```